cpp/include/raft/distance/detail/distance.cuh

```diff
@@ -92,7 +93,8 @@ template <raft::distance::DistanceType distanceType,
           typename FinalLambda,
           typename Index_>
 struct DistanceImpl {
-  void run(const InType* x,
+  void run(raft::device_resources const& handle,
```

Comment:
One thing you could do is even go a step further and just have this accept `raft::resources`. Then we can use the `raft::resources::get_stream(handle)` accessor function without having to import anything other than the stream.

Reply:
done in the last commit